internal/client.go

```diff
@@ -31,23 +34,52 @@ func DiscoverContextURL(ctx context.Context, service, domain string) (string, er
 	}
 
 	if len(addrs) == 0 {
-		return "", fmt.Errorf("webdav: domain doesn't have an SRV record")
+		return "", errors.New("webdav: domain doesn't have an SRV record")
 	}
 	addr := addrs[0]
 
 	target := strings.TrimSuffix(addr.Target, ".")
 	if target == "" {
-		return "", fmt.Errorf("webdav: empty target in SRV record")
+		return "", errors.New("webdav: empty target in SRV record")
 	}
 
-	// TODO: perform a TXT lookup, check for a "path" key in the response
-	u := url.URL{Scheme: "https"}
+	txtName := fmt.Sprintf("_%ss._tcp.%s", service, domain)
+	txtRecords, err := resolver.LookupTXT(ctx, txtName)
+	if dnsErr, ok := err.(*net.DNSError); ok {
+		if dnsErr.IsTemporary {
+			return "", err
+		}
+	} else if err != nil {
+		return "", err
+	}
+
+	var path string
+	switch len(txtRecords) {
+	case 0:
+		path = "/.well-known/" + service
+	case 1:
+		record := txtRecords[0]
+		if !strings.HasPrefix(record, "path=") {
+			break
```

Comment:
This leaves `path` empty. Probably we should return an error instead?

Reply:
Absolutely!